litestar/handlers/base.py

```diff
@@ -431,76 +496,19 @@ def _validate_dependency_is_unique(dependencies: dict[str, Provide], key: str, p
                     f"If you wish to override a provider, it must have the same key."
                 )
 
-    def on_registration(self, app: Litestar) -> None:
```

Comment:
Why this breaking change?

Reply:
It's not a breaking change. This is internal API. It's not meant to be user facing at all. 

Anyhow, we do not need to inject the app during registration at all, and the stuff there was hacky.

Reply:
It is a breaking change. This API is public and documented.

https://docs.litestar.dev/2/reference/handlers.html#litestar.handlers.BaseRouteHandler.on_registration

If we the changes you made make usage of this parameter obsolete than you can simply not use it. No need to possibly break things for users.

Reply:
ok